docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+    "from qiskit_machine_learning.datasets.dataset_helper import discretize_and_truncate\n",
+    "from qiskit_machine_learning.neural_networks import CircuitQNN\n",
+    "\n",
+    "# Set seed for random generators\n",
+    "torch.manual_seed(42)\n",
+    "algorithm_globals.random_seed = 42"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Data and Representation\n",
+    "\n",
+    "First, we need to load our training data $X$.\n",
+    "\n",
+    "In this tutorial, the training data is given by samples from a 2D multivariate normal distribution.\n",
+    "\n",
+    "The trained generator corresponds to an $n$-qubit quantum state\n",
+    "\\begin{equation}\n",
+    "|g_{\\text{trained}}\\rangle=\\sum\\limits_{j=0}^{k-1}\\sqrt{p_{j}}|x_{j}\\rangle,\n",
+    "\\end{equation}\n",
+    "where the basis states $|x_{j}\\rangle$ represent the data items in the training data set\n",
+    "$X={x_0, \\ldots, x_{k-1}}$ with $k\\leq 2^n$ and $p_j$ refers to the sampling probability\n",
+    "of $|x_{j}\\rangle$.\n",
+    "\n",
+    "To facilitate this representation, we need to map the samples from the multivariate\n",
+    "normal distribution to discrete values. The number of values that can be represented\n",
+    "depends on the number of qubits used for the mapping.\n",
+    "Hence, the data resolution is defined by the number of qubits.\n",
+    "If we use $3$ qubits to represent one feature, we have $2^3 = 8$ discrete values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Load the training data\n",
+    "training_data = algorithm_globals.random.multivariate_normal(\n",
+    "    mean=[0.0, 0.0], cov=[[1, 0], [0, 1]], size=1000, check_valid=\"warn\", tol=1e-8, method=\"svd\"\n",
+    ")\n",
+    "# Define minimal and maximal values for the training data\n",
+    "bounds_min = np.percentile(training_data, 5, axis=0)\n",
+    "bounds_max = np.percentile(training_data, 95, axis=0)\n",
+    "bounds = []\n",
+    "for i, _ in enumerate(bounds_min):\n",
+    "    bounds.append([bounds_min[i], bounds_max[i]])\n",
+    "\n",
+    "# Determine data resolution for each dimension of the training data in terms\n",
+    "# of the number of qubits used to represent each data dimension.\n",
+    "data_dim = [3, 3]\n",
+    "\n",
+    "# Pre-processing, i.e., discretization of the data (gridding)\n",
+    "(training_data, _, grid_elements, prob_data) = discretize_and_truncate(\n",
+    "    training_data,\n",
+    "    np.asarray(bounds),\n",
+    "    data_dim,\n",
+    "    return_data_grid_elements=True,\n",
+    "    return_prob=True,\n",
+    "    prob_non_zero=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
```

Comment:
Just a thought, it might be nice to plot the training data at this stage to show what distribution we are supposed to reproduce

Reply:
Good idea, let me ask @Zoufalc :)